### ISSUE TITLE
Simple and Scalable Strategies to Continually Pre-train Large Language Models

### ISSUE THREAD

**Issue by Hiroki11x** [Hiroki11x/Papers#428]
## Links
https://arxiv.org/pdf/2403.08763

## Author
Mila, Irina や Adam

### 要約
本論文は，大規模言語モデル（LLM）の**継続的事前学習（continual pre-training）**を効率的に行うための単純かつスケーラブルな戦略を提案する。  
これにより、従来の再学習と同等の性能を維持しつつ、**計算コストを大幅に削減**できることを実証している。

---

## 🧭 概要と研究目的
従来、LLMは新しいデータが得られるたびに**ゼロから再学習**されてきた。しかしこの方法は膨大な計算資源を要する。本研究では、既存モデルを「継続的に」アップデートする**Continual Pre-training**を提案し、再学習と同等の性能を維持できるかを大規模実験で検証している。

---

## 🧩 方法と新規性
本研究の中核は、**3つの単純な手法の組み合わせ**である：
1. **Learning Rate (LR) Re-warming** — 新データ学習前に学習率を再上昇させる  
2. **LR Re-decaying** — 学習後半で再び学習率を減衰させる  
3. **Replay of Previous Data** — 旧データを少量（1〜25%）再提示し、忘却を防止  

これらを組み合わせることで、分布シフト（英語→英語、英語→ドイツ語）の両方で、**フル再学習とほぼ同等の性能**を達成した点が新規性である。特に10Bパラメータ規模のLLMにおいてもこの手法が機能することを示した。

---

## 🧠 実験設計
- **モデル:** GPT-NeoXベースのデコーダオンリーTransformer（405M・10Bパラメータ）  
- **データ:** The Pile, SlimPajama, German Common Crawl（計数百B tokens）  
- **評価:** 英語およびドイツ語ベンチマーク（MMLU, HellaSwag, TriviaQA など）  

**設定:**
- 弱シフト（英語→英語）と強シフト（英語→ドイツ語）の2条件  
- 比較対象は「完全再学習（D₀∪D₁）」と「継続学習（D₀→D₁）」  

---

### Figure 1（p.3）
- **左:** 再学習に比べ継続学習は必要計算量が半減  
- **中央:** 最終検証損失はほぼ同等  
- **右:** 評価ベンチマーク平均スコアもほぼ一致  

<img width="674" height="413" alt="Image" src="https://github.com/user-attachments/assets/15114176-bb51-4a97-bc34-a76e8493659d" />

➡️ *結論:* 「LR再ウォーム＋再デケイ＋少量リプレイ」で再学習と同等の性能を、**半分以下の計算コスト**で達成。

---

### Figure 4（p.12）
- LRの再ウォーム・再デケイを行わない場合、新データへの適応が弱い。  
- 再ウォーム値を上げすぎると忘却が進むが、適度な範囲（1.5–3×10⁻⁴）で良好なトレードオフ。  

<img width="644" height="524" alt="Image" src="https://github.com/user-attachments/assets/fe602b8e-b14c-4101-a166-80753225e2bd" />

➡️ *示唆:* 「学習率の再加熱」が新データ適応に不可欠。

---

### Figure 5（p.14）
- **1〜5%のリプレイ**でも忘却が大きく軽減。  
- 強い分布シフト（英→独）では25%程度のリプレイが最適。  

<img width="648" height="512" alt="Image" src="https://github.com/user-attachments/assets/ddeb6cec-5161-473a-b1ae-e15fe8274e4f" />

➡️ *示唆:* わずかな旧データ再提示で知識保持を実現。

---

### Figure 7（p.18）
- **10Bモデルでも405Mモデルと同様の傾向**が確認され、スケールに依存しない。  
- リプレイを用いた継続学習モデルは、完全再学習（600B tokens）と平均損失が0.02差。

<img width="646" height="548" alt="Image" src="https://github.com/user-attachments/assets/10cb1251-f2f9-4dbc-9391-a5262879e33c" />

---

### Figure 9–10（p.23）
- **新手法「Infinite LR Schedule」**を提案：再ウォーム不要で学習を継続できるスケジュール。  
- Cosine decayと同等の性能を維持しつつ、タスク間で滑らかに遷移。  

➡️ *重要な貢献:* 「固定トークン予算に依存しないスケジュール設計」を提示。

---

## 🧪 貢献の実証方法
- 弱／強シフトでの比較実験（Pile→SlimPajama, Pile→German）  
- 異なるモデルサイズ（405M, 10B）での再現性検証  
- LRスケジュールとリプレイ率のアブレーション実験  
- 英独両言語ベンチマークでの性能評価  

すべての実験で、**継続学習が完全**
